### PR TITLE
[DNM] Use geom column in Spatial Queries

### LINKED ISSUE #34: Full table scans while using spatial index?
I've a table created with Koop which contains around 65000 items. My performance was very low, even when I zoomed in to an area with a low amount of features.

I had a close look at the pgcache provider and after a second check on my database I found out the geom column is not used at all? Can you explain why you chose to do spatial part in the feature column as well, because I cannot understand why at this moment.

I analyzed one of the queries with the postgres explain command and got the following result:
EXPLAIN  select id, feature->>'properties' as props, feature->>'geometry' as geom from "WFS:rce:rce:NationalListedMonuments:0" WHERE ST_GeomFromGeoJSON(feature->>'geometry') && ST_SetSRID('BOX3D(5.4217529140341885 53.17641168037742,5.427246078096215 53.1797038995227)'::box3d,4326) ORDER BY id; 

"Sort  (cost=11127.26..11159.04 rows=12711 width=36)"
"  Sort Key: id"
"  ->  Seq Scan on "WFS:rce:rce:NationalListedMonuments:0"  (cost=0.00..10260.77 rows=12711 width=36)"
"        Filter: (st_geomfromgeojson((feature ->> 'geometry'::text)) && '0103000020E610000001000000050000000214F3FEDFAF15403DED6EA894964A400214F3FEDFAF1540F3B4918900974A40ED11F3FE7FB51540F3B4918900974A40ED11F3FE7FB515403DED6EA894964A400214F3FEDFAF15403DED6E (...)"

Then I refactored the whole pgcache provider in order to use the geometry column for my geometries and I got this result:
EXPLAIN  select id, feature->>'properties' as props, feature->>'geometry' as geom from "WFS:rce:rce:NationalListedMonuments:0" WHERE geom && ST_SetSRID('BOX3D(5.4217529140341885 53.17641168037742,5.427246078096215 53.1797038995227)'::box3d,4326) ORDER BY id; 
"Sort  (cost=59.95..59.98 rows=14 width=36)"
"  Sort Key: id"
"  ->  Bitmap Heap Scan on "WFS:rce:rce:NationalListedMonuments:0"  (cost=4.39..59.68 rows=14 width=36)"
"        Recheck Cond: (geom && '0103000020E610000001000000050000000214F3FEDFAF15403DED6EA894964A400214F3FEDFAF1540F3B4918900974A40ED11F3FE7FB51540F3B4918900974A40ED11F3FE7FB515403DED6EA894964A400214F3FEDFAF15403DED6EA894964A40'::geometry)"
"        ->  Bitmap Index Scan on wfsrcercenationallistedmonuments0_gix  (cost=0.00..4.39 rows=14 width=0)"
"              Index Cond: (geom && '0103000020E610000001000000050000000214F3FEDFAF15403DED6EA894964A400214F3FEDFAF1540F3B4918900974A40ED11F3FE7FB51540F3B4918900974A40ED11F3FE7FB515403DED6EA894964A400214F3FEDFAF15403DED6EA894964A40'::geometry)"

I'm still using the geometry from the feature field, but doing all the smart spatial stuff on the geometry column

The calculations were done on different systems, but on the same data, but I think the differences are clear.

The first result was on a Windows Server 2012R2 with 8GB RAM and 4 CPU cores
the second result was on my development laptop running Windows 10 with 8GB Ram and 2 CPU cores with hyperthreading (4 total).

I can make a pull request from my pgcache using the geometry column, but this will be a huge change for all existing systems and databases. Therefore I think it's better to have a discussion about this first before I make this PR.

### PR DESCRIPTION
**Pull request by dmfenton**
This brings in @peacenlove's work to speed up spatial queries by utilizing the geometry column in the db.

cc @koopjs/dev

Resolves #34

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -325,6 +324,7 @@ module.exports = {
     var select = self._buildQuery(id, options)
     self.log.debug('Selecting data', select)
     self._query(select, function (err, result) {
+      console.log(select, err)
```
Comment (ungoldman) [inline index.js]: errant console.log

**Comment by ungoldman** [comment]
everything looks sensible to me here. I'll remove the console log then merge it.

**Comment by ungoldman** [comment]
@dmfenton can you sanity check https://github.com/koopjs/koop-pgcache/commit/148b7ff0292b660495508f67c1a1dd81910bb2c0, then I think this is good to go.

**Comment by dmfenton** [comment]
Given that this is going to require a migration I'd like to add that to this PR before merging. So DNM please.

**Comment by dmfenton** [comment]
This has been rolled into the 2.0 branch #47

**Pull request closed without merging**